services/ymax-planner/src/watchers/cctp-watcher.ts

```diff
@@ -129,6 +131,13 @@ export const watchCctpTransfer = ({
         resolve(false);
       }
     }, timeoutMs);
+
+    if (signal) {
+      signal.addEventListener('abort', () => {
+        cleanup();
+        resolve(false);
+      });
+    }
```

Comment:
Please do this immediately after defining `cleanup`, and also consider a simplification:
```suggestion
    signal?.addEventListener('abort', () => {
      cleanup();
      resolve(false);
    });
```
or better yet, refactor `cleanup` into a `finish` function that wraps `resolve` to include cleanup:
```ts
    const finish = (result: boolean) => {
      resolve(result);
      if (timeoutId) clearTimeout(timeoutId);
      for (const { event, listener } of listeners) {
        void provider.off(event, listener);
      }
      listeners = [];
    };
    signal?.addEventListener('abort', () => finish(false));
```